lomond/session.py

```diff
@@ -255,17 +293,19 @@ def run(self,
 
         # Connected to the server, but not yet upgraded to websockets
         yield events.Connected(url)
-        self._poll_start = time.time()
 
+        ready = False
         try:
             while not websocket.is_closed:
                 reads, _errors = self._select(sock, poll)
 
-                # Check for polls / pings
-                for event in self._regular(poll,
-                                           ping_rate,
-                                           close_timeout):
-                    yield event
+                if ready:
```

Comment:
The 'if ready' ensures that polls and timeouts only occur when the websocket is in a ready state.